merkletree.go

```diff
@@ -121,11 +121,11 @@ func computePrivateIndex(key string) []byte {
 	return crypto.Digest([]byte(key))
```

Comment:
I also think it makes sense to make this a separate PR. Will do so in the following days, OK?